config/src/runtime_upgrades.rs

```diff
@@ -0,0 +1,265 @@
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: serde::Deserializer<'de>,
+        {
+            Ok(Self(AtomicCell::new(T::deserialize(deserializer)?)))
+        }
+    }
+
+    impl<T: Serialize + Clone + Copy> Serialize for Atomic<T> {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: serde::Serializer,
+        {
+            (self.0.load()).serialize(serializer)
+        }
+    }
+
+    // -----------------------------------------------------------------------
+
+    /// Structure that encapsulates a configuration value as well as a handle
+    /// for reloading other parts of the program.  This is the `struct` that you
+    /// want to use 99% of the time.
+    pub struct Reconfigure<T: Clone + Copy, H: Reload<T>>(Atomic<T>, H);
+
+    impl<T> Default for Reconfigure<T, Singleton<T>>
+    where
+        T: Default + Clone + Copy + Send + Sync + Debug,
+    {
+        fn default() -> Self {
+            Self(Atomic::default(), Singleton::default())
+        }
+    }
+
+    impl<T: Clone + Copy, H: Reload<T>> Reload<T> for Reconfigure<T, H> {
+        fn reload(&self, item: T) -> Result<()> {
+            self.1.reload(item)?;
+            self.0.reload(item)
+        }
+    }
+
+    impl<T: Serialize + Clone + Copy, H: Reload<T>> Serialize for Reconfigure<T, H> {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: serde::Serializer,
+        {
+            // We only want the actual (simple) value to be part of the serializing
+            self.0.serialize(serializer)
+        }
+    }
+
+    impl<'de, T: Deserialize<'de> + Copy + Clone, H: Reload<T> + Default> Deserialize<'de>
+        for Reconfigure<T, H>
+    {
+        fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: serde::Deserializer<'de>,
+        {
+            Ok(Self(Atomic::<T>::deserialize(deserializer)?, H::default()))
+        }
+    }
+
+    impl<T: Clone + Copy, H: Reload<T> + Clone> Clone for Reconfigure<T, H> {
+        fn clone(&self) -> Self {
+            Self(self.0.clone(), self.1.clone())
+        }
+    }
+
+    impl<T: Clone + Copy + Debug, H: Reload<T> + Debug> Debug for Reconfigure<T, H> {
+        fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+            f.debug_tuple("Reconfigure")
+                .field(&self.0)
+                .field(&self.1)
+                .finish()
+        }
+    }
+
+    impl<T: Clone + Copy, H: Reload<T>> Reconfigure<T, H> {
+        /// Getter for value wrapped by [`Reconfigurable`]
+        pub fn value(&self) -> T {
+            self.0 .0.load()
+        }
+    }
+
+    impl<T: Clone + Copy + Send + Sync + Debug> Reconfigure<T, Singleton<T>> {
+        /// Set the [`SideEffect`] handle
+        ///
+        /// # Errors
+        /// If [`SideEffect::handle`] fails.
```

Comment:
It is no longer named SideEffect. There are several places where this needs to be replaced, if I understood it correctly.